actions/views/root.js

```diff
@@ -48,20 +48,29 @@ export function unregisterPluginTranslationsSource(pluginId) {
 }
 
 export function loadTranslations(locale, url) {
-    return (dispatch) => {
-        const translations = {};
+    return async (dispatch) => {
+        const translations = {...en};
         Object.values(pluginTranslationSources).forEach((pluginFunc) => {
             Object.assign(translations, pluginFunc(locale));
         });
 
-        // No need to go to the server for EN
-        if (locale === 'en') {
-            copyAndDispatchTranslations(dispatch, translations, en, locale);
-            return;
+        // Need to go to the server for languages other than English
+
+        if (locale !== 'en') {
+            try {
+                const serverTranslations = await Client4.getTranslations(url);
+                Object.assign(translations, serverTranslations);
+            } catch (error) {
+                console.error(error); //eslint-disable-line no-console
+            }
         }
-        Client4.getTranslations(url).then((serverTranslations) => {
-            copyAndDispatchTranslations(dispatch, translations, serverTranslations, locale);
-        }).catch(() => {}); // eslint-disable-line no-empty-function
+        dispatch({
+            type: ActionTypes.RECEIVED_TRANSLATIONS,
+            data: {
+                locale,
+                translations,
+            },
+        });
     };
 }
 
```

Comment:
you can also remove the `copyAndDispatchTranslations` function on line 77
https://github.com/mattermost/mattermost-webapp/pull/4561/files#diff-b756d78f088056823811c9d55e18baefR77

Reply:
It's also called on line 41 https://github.com/mattermost/mattermost-webapp/blob/27b7997805a9dfc086d4518ee6dedf9ee16dba07/actions/views/root.js#L41

Reply:
@ali-farooq0 this merge is still blocked due to this, can you mark this as resolved?

Reply:
Apologies for the delay @davidjwilkins. Ahh didn't realize we were using `copyAndDispatchTranslations` in another spot. I would recommend either getting rid of `copyAndDispatchTranslations` and simply dispatching inline on line 41, or use `copyAndDispatchTranslations` on line 66 to be consistent. Will leave it upto you to decide :), either one works for me.